app/src/main/java/org/covidwatch/android/presentation/test/PotentialRiskFragment.kt

```diff
@@ -0,0 +1,72 @@
+package org.covidwatch.android.presentation.test
+
+import android.app.DatePickerDialog
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.view.isVisible
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.Observer
+import androidx.navigation.fragment.findNavController
+import org.covidwatch.android.R
+import org.covidwatch.android.databinding.FragmentPotentialRiskBinding
+import org.covidwatch.android.databinding.FragmentTestQuestionsBinding
+import java.util.*
+
+class PotentialRiskFragment : Fragment() {
+
+    private var _binding: FragmentPotentialRiskBinding? = null
+    private val binding get() = _binding!!
+
+    private val testQuestionsViewModel: TestQuestionsViewModel by viewModels()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        _binding = FragmentPotentialRiskBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        testQuestionsViewModel.isTested.observe(viewLifecycleOwner, Observer {
+            updateUi(it)
+        })
+        initClickListeners()
+    }
+
+    private fun initClickListeners() {
+        binding.closeButton.setOnClickListener {
+            findNavController().popBackStack()
+        }
+        binding.riskNegativeButton.setOnClickListener {
+            testQuestionsViewModel.onRadioButtonClicked(false)
+        }
+        binding.riskPositiveButton.setOnClickListener {
+            testQuestionsViewModel.onRadioButtonClicked(true)
```

Comment:
in order for it to work we need to observe the rest of the `LiveData`s too, along with logic that comes with it

Reply:
but we're repeating ourselves here, and I don't have a solution right now

Reply:
I duplicated the logic from TestQuestionsFragment into PotentialRiskFragment.  I also abstracted out the date picker logic into a mixin, enabling some code re-use.  I wasn't able to do any more than that however because the listeners and observers are tightly coupled to the UI bindings, preventing dependency injection into a clean, default interface function implementation.